src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs

```diff
@@ -72,6 +83,19 @@ public MessageMetadata GetMessageMetadata(string messageTypeIdentifier)
             return null;
         }
 
+        Type GetType(string messageTypeIdentifier)
```

Comment:
Yeah but one "atomic" the other one not that's why GetOrAdd should be
preferred

On Monday, August 29, 2016, Brandon Ording <notifications@github.com> wrote:

> In src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs
> <https://github.com/Particular/NServiceBus/pull/4066#discussion_r76643346>
> :
>
> > @@ -72,6 +83,19 @@ public MessageMetadata GetMessageMetadata(string messageTypeIdentifier)
> >              return null;
> >          }
> >
> > +        Type GetType(string messageTypeIdentifier)
>
> I suppose it could have been used, but I didn't think about it. From a
> perf standpoint, both approaches appear to be the same.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/4066/files/c3053323395c25bdc0b43cdda48dd6751d3969bb#r76643346>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAKosgAx2-T8xKc9Do186VarvHaRO3I-ks5qkw72gaJpZM4Jl9wF>
> .
>